go/vt/vtgate/planbuilder/vindex_func.go

```diff
@@ -132,8 +130,8 @@ func (vf *vindexFunc) SupplyCol(col *sqlparser.ColName) (rc *resultColumn, colNu
 }
 
 // SupplyWeightString implements the logicalPlan interface
-func (vf *vindexFunc) SupplyWeightString(colNumber int) (weightcolNumber int, err error) {
-	return 0, errors.New("cannot do collation on vindex function")
+func (vf *vindexFunc) SupplyWeightString(colNumber int) (weightcolNumber int) {
+	return -1
```

Comment:
hmm... why remove the error return value if we sometimes want to signal an error?

Reply:
when we dont want weight strings we can just return -1 from the function and it wont add weight string. There wasn't any place were we would want to throw an error anymore since this is going to be the default behaviour so I removed the error return value

Reply:
changed back in last commit